wordpress-mobile/WordPress-iOS
# MediaLibrary: MediaURLExporter and tests.

Continuing from #7118, this PR introduces the `MediaURLExporter` which primarily handles the detection of what the file at the URL actually is and exporting as needed.

Similarly to `MediaImageExporter`, errors are propagated via `throw` to keep track of exactly what went wrong along the way.

This exporter uses a value type for `URLExport` as the resulting export for the file depends on type of file it actually was.

- For images, we let the `MediaImageExporter` handle the export, if detected.
- For videos, we reuse the same kind code seen from the current export implementation in `develop` via `AVAssetExportSession`.
- For gifs, we just copy the file to our ideal Media directory.

Note: this exporter will be handy with the upcoming share extension.

To test:
1. Make sure the code compiles.
2. Run the tests for `MediaURLExporterTests.swift`

Needs review: @frosty and @SergioEstevao round four, final round 🤕😄
